evm_arithmetization/src/generation/state.rs

```diff
@@ -212,7 +212,14 @@ pub(crate) trait State<F: Field> {
                     if !running {
                         assert_eq!(self.get_clock() - final_clock, NUM_EXTRA_CYCLES_AFTER - 1);
                     }
-                    let final_mem = self.get_full_memory();
+                    let final_mem = if let Some(mut mem) = self.get_full_memory() {
+                        for &ctx in &self.get_generation_state().pruned_contexts {
+                            mem.contexts[ctx] = MemoryContextState::default();
+                        }
+                        Some(mem)
+                    } else {
+                        None
```

Comment:
For my education, what is the case when `get_full_memory` is `None`? What does that mean that `GenerationState` use the default impl and returns `None`?

Reply:
When you're generating the proof (with the full trace), you don't need this memory. The `Memory` trace will be generated from the list of `MemOps` instead.